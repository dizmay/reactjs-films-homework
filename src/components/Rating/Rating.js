/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';
import styles from './Rating.scss';

const Rating = () => (
  <div className={styles.rateContainer}>
    <div className={styles.rate}>
      <input type="radio" id="star5" />
      <label htmlFor="star5" />
      <input type="radio" id="star4" />
      <label htmlFor="star4" />
      <input type="radio" id="star3" />
      <label htmlFor="star3" />
      <input type="radio" id="star2" />
      <label htmlFor="star2" />
      <input type="radio" id="star1" />
      <label htmlFor="star1" />
    </div>
    <div className={styles.averageRate}>4.8</div>
  </div>
);

export default Rating;
