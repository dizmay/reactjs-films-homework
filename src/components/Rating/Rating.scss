* {
  margin: 0;
  padding: 0;
}

.rateContainer {
  display: flex;
  align-items: center;
}

.rate {
  float: left;
  height: 3rem;
  padding: 0 0.6rem;
}

.rate:not(:checked) > input {
  position: absolute;
  top: -9999px;
}

.rate:not(:checked) > label {
  float: right;
  width: 1em;
  overflow: hidden;
  white-space: nowrap;
  cursor: pointer;
  font-size: 1.875rem;
  color: #ccc;
}

.rate:not(:checked) > label::before {
  content: 'â˜… ';
}

.rate > input:checked ~ label {
  color: #00a3f5;
}

.rate:not(:checked) > label:hover,
.rate:not(:checked) > label:hover ~ label {
  color: #00a3f5;
}

.rate > input:checked + label:hover,
.rate > input:checked ~ label:hover,
.rate > input:checked + label:hover ~ label,
.rate > input:checked ~ label:hover ~ label,
.rate > label:hover ~ input:checked ~ label {
  color: #00a3f5;
}

.averageRate {
  color: #fff;
  font-size: 0.9rem;
  border: 0.15rem solid #00a3f5;
  border-radius: 0.2rem;
  padding: 0.1rem 0.4rem;
}
