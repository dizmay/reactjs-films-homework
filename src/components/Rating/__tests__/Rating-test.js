import React from 'react';
import renderer from 'react-test-renderer';
import Rating from '../Rating';

it('renders correctly', () => {
  const tree = renderer.create(<Rating />).toJSON();
  expect(tree).toMatchSnapshot();
});
