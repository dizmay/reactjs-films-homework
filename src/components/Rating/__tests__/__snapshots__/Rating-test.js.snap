// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="rateContainer"
>
  <div
    className="rate"
  >
    <input
      id="star5"
      type="radio"
    />
    <label
      htmlFor="star5"
    />
    <input
      id="star4"
      type="radio"
    />
    <label
      htmlFor="star4"
    />
    <input
      id="star3"
      type="radio"
    />
    <label
      htmlFor="star3"
    />
    <input
      id="star2"
      type="radio"
    />
    <label
      htmlFor="star2"
    />
    <input
      id="star1"
      type="radio"
    />
    <label
      htmlFor="star1"
    />
  </div>
  <div
    className="averageRate"
  >
    4.8
  </div>
</div>
`;
